golang/internal/grpc/grpc.go

```diff
@@ -289,6 +309,33 @@ func grpcLoop(
 		command := new(agent.AgentCommand)
 		err = stream.RecvMsg(command)
 		if err != nil {
+			s := status.Convert(err)
+			if s != nil && (s.Code() == codes.Unauthenticated || s.Code() == codes.PermissionDenied || s.Code() == codes.NotFound) {
+				if l.AppConfig.GrpcToken.Type == config.Connection {
+					log.Error().Err(err).Msg("Invalid connection token. Removing")
+
+					// overwrite jwt token
+					err := l.Secrets.SaveConnectionToken("")
+					if err != nil {
+						log.Err(err).Msg("Failed to delete the invalid connection token. Exiting")
+
+						os.Exit(1)
```

Comment:
Missing the debug message like below.

Reply:
It is present on the error log up above.